### PR TITLE
Update hive-caching documentation for Rubix 0.3.16

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

File: presto-docs/src/main/sphinx/connector/hive-caching.rst [hunk 1]
```diff
@@ -177,16 +177,17 @@ The implementation of the cache exposes a `number of metrics
...
+The following example query returns the total amount of data fetched from source
+filesystem by all Presto nodes:
 
 .. code-block:: sql
 
-  SELECT fs.readfromremote + async_downloaded_mb.count
-  FROM jmx.current."rubix:catalog=hive,name=stats" fs,
-       jmx.current."metrics:name=rubix.bookkeeper.count.async_downloaded_mb" async_downloaded_mb
-  WHERE fs.node = async_downloaded_mb.node;
+  SELECT sum(mb_read_from_source) FROM jmx.current."rubix:catalog=hive,name=stats";
```
Comment (sopel39) [inline presto-docs/src/main/sphinx/connector/hive-caching.rst:190]: this is only valid for read-though mode, for async mode, we still need to account for `jmx.current."metrics:name=rubix.bookkeeper.count.async_downloaded_mb" `

https://github.com/qubole/rubix/pull/438 will introduce `total_system_source_mb_read` global metrics

Comment (losipiuk) [reply presto-docs/src/main/sphinx/connector/hive-caching.rst]: Thanks. I chaned the example query to return average cache hit. I would rather not mix-in `metrics:name=rubix.bookkeeper.count.async_downloaded_mb` into example as it does not play well with presto (no catalog name in key).

Comment (sopel39) [reply presto-docs/src/main/sphinx/connector/hive-caching.rst]: > I would rather not mix-in metrics:name=rubix.bookkeeper.count.async_downloaded_mb into example as it does not play well with presto (no catalog name in key).

Right, `cache_hit` is fine, yet it doesn't tell the whole store. How do you know if Presto is not going directly to data source?

Comment (losipiuk) [reply presto-docs/src/main/sphinx/connector/hive-caching.rst]: We do not. But I think the bottom line is that we just cannot reasonably provide this information until qubole/rubix#438.
Well we can as long as only single connector uses caching, but I do not want to elaborate too much about that in the documentation.

I would rather push qubole/rubix#438 and update the documentation with example regarding "data read from source" as it gets merged and we update rubix.

File: presto-docs/src/main/sphinx/connector/hive-caching.rst [hunk 2]
```diff
@@ -177,16 +177,16 @@ The implementation of the cache exposes a `number of metrics
 :doc:`inspect these and other metrics directly in Presto with the JMX connector
 or in external tools </admin/jmx>`.
 
-The following example query returns the total amount of data fetched by the
-BookKeeper on the coordinator and all the separate nodes on the workers in the
-default async mode:
+If you are using JMX connector, basic caching statistics are exposed via
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/connector/hive-caching.rst:180]: Remove "If you are using JXM connector" sins this is already said earlier.

Basic caching statistics are available in the ...

File: presto-docs/src/main/sphinx/connector/hive-caching.rst [hunk 3]
```diff
@@ -177,16 +177,16 @@ The implementation of the cache exposes a `number of metrics
...
+``jmx.current."rubix:catalog=<catalog_name>,name=stats"`` table.
+You can read more detailed statistics by querying ``jmx.current."rubix:catalog=<catalog_name>,name=stats``.
+You need to replace ``<catalog_name>`` with name of you Hive catalog.
+
+The following example query returns average cache hit ratio for given Hive catalog.
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/connector/hive-caching.rst:185]: The following example query returns the average cache hit ratio for the ``hive`` catalog:

File: presto-docs/src/main/sphinx/connector/hive-caching.rst [hunk 4]
```diff
@@ -177,16 +177,16 @@ The implementation of the cache exposes a `number of metrics
...
 
 .. code-block:: sql
 
-  SELECT fs.readfromremote + async_downloaded_mb.count
-  FROM jmx.current."rubix:catalog=hive,name=stats" fs,
-       jmx.current."metrics:name=rubix.bookkeeper.count.async_downloaded_mb" async_downloaded_mb
-  WHERE fs.node = async_downloaded_mb.node;
+  SELECT avg(cache_hit) FROM jmx.current."rubix:catalog=hive,name=stats" where not is_nan(cache_hit);
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/connector/hive-caching.rst:189]:
```
SELECT avg(cache_hit) 
FROM jmx.current."rubix:catalog=hive,name=stats" 
WHERE NOT is_nan(cache_hit);
```

File: presto-docs/src/main/sphinx/connector/hive-caching.rst (hunk 1 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/connector/hive-caching.rst]: That sounds reasonable to me.

File: presto-docs/src/main/sphinx/connector/hive-caching.rst (hunk 2 above)
Comment (losipiuk) [reply presto-docs/src/main/sphinx/connector/hive-caching.rst:180]: Should it be 
```
The table ``jmx.current."rubix:catalog=<catalog_name>,name=stats`` 
```
Or 
```
The ``jmx.current."rubix:catalog=<catalog_name>,name=stats`` table
```

Latter sound more proper to me, but I may be very wrong here :)

File: presto-docs/src/main/sphinx/connector/hive-caching.rst (hunk 3 above)
Comment (losipiuk) [reply presto-docs/src/main/sphinx/connector/hive-caching.rst:185]: A hint @mosabua: 

If you are suggesting rewording of whole paragraphs you can use suggestion mechanism of Github code review. 
To do so:
* mark the lines for which you want to suggest alternative.
* click comment
* click +/- button in the comment editor
* provide suggestion.

Screenshot: 
image

**Comment by losipiuk** [comment]
Addressed comments.

File: presto-docs/src/main/sphinx/connector/hive-caching.rst (hunk 2 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/connector/hive-caching.rst]: either is fine

**Pull request merged**